pkg/compactor/compactor.go

```diff
@@ -381,41 +381,26 @@ func (c *Compactor) running(ctx context.Context) error {
 	}
 }
 
-func (c *Compactor) compactUsersWithRetries(ctx context.Context) {
-	retries := util.NewBackoff(ctx, util.BackoffConfig{
-		MinBackoff: c.compactorCfg.retryMinBackoff,
-		MaxBackoff: c.compactorCfg.retryMaxBackoff,
-		MaxRetries: c.compactorCfg.CompactionRetries,
-	})
-
+func (c *Compactor) compactUsers(ctx context.Context) (returnErr error) {
```

Comment:
Returned error is not used anywhere. We should remove it (we already log all errors anyway), and perhaps also simplify `returnErr` to boolean flag for updating metrics.

Reply:
Does it look better now?

Reply:
Looks better. As a final change, I'd suggest removing unused return value completely, and convert `succeeded` into local variable.

Reply:
Oh yeah, sure, done.